File: Question5.cs
using System;
namespace Algorithm_Practice {
    public class Question5 {
        public Question5() {
        }

        public void Action(int n) {
            Console.WriteLine("Hanoi {0:d}", n);
            Hanoi(n, "A", "B", "C");
        }

        private void Hanoi(int n, String left, String center, String right) {
            if (n > 0) {
                Hanoi(n - 1, left, right, center);
                Console.WriteLine("{0:s} to {1:s}", left, right);
                Hanoi(n - 1, center, left, right);
            }
        }
    }
}

File: Question5.java
class Question5 {

	public static void main(String[] args) {
		// TODO 自動生成されたメソッド・スタブ
		final int N = 4;
		System.out.println("Hanoi " + N);
		hanoi (N, "A", "B", "C");
	}

	static void hanoi (int n, String left, String center, String right) {
		if (n > 0) {
			hanoi(n-1, left, right, center);
			System.out.println(left + " -> " + right);
			hanoi(n-1, center, left, right);
		}
	}
}
